client/src/App.js

```diff
@@ -1,13 +1,75 @@
 import "./App.css";
+import React, { useState, useEffect } from "react";
+import AddVideoForm from "./AddVideoForm";
+import moment from "moment";
+import VideoComponent from "./VideoComponent";
+import Footer from "./Footer";
+
+const App = () => {
+  const [videoList, setVideoList] = useState([]);
+  const [sortedVideoList, setSortedVideoList] = useState([]);
+  useEffect(() => {
+    fetchData();
+  }, []);
+
+  const fetchData = async () => {
+    try {
+      const res = await fetch("http://localhost:5000/videos");
+      if (res.ok) {
+        const data = await res.json();
+        setVideoList(data);
+        console.log(data);
+        const sortedList = [...data].sort((a, b) => b.rating - a.rating);
+        setSortedVideoList(sortedList);
+      } else {
+        throw new Error("Failed to fetch videos");
+      }
+    } catch (error) {
+      console.error("Error fetching videos:", error);
+    }
+  };
```

Comment:
This could all be a custom hook:
```
const useFetchData = () => {
  const [videoList] ...
  const [sortedVideoList ]...

 const fetchData = () ....

 useEffect(() => {
    fetchData()
  }, [])

return{[videoList, sortedVideoList}
}

const App = () => {
  const {videoList, sortedVideoList} = useFetchData()
}
```

This keeps the logic separte from the component and allows the component to focus only on displaying data